### ISSUE TITLE
UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. 

### ISSUE THREAD

**Issue by jiyuwangbupt** [ultralytics/ultralytics#283]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. 

class_loss is inf, box_loss is 0, dfl_loss is 0.

By the way, this environment runs fine on yolov5 without warnings. The configuration of the YOLOv8 environment is based on the yolov5 environment, and then use pip install ultralytics (no errors were generated during the period).
### Environment

Ultralytics YOLOv8.0.3 🚀 Python-3.8.5 torch-1.11.0+cu102 CUDA:0 (Tesla P40, 12288MiB)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Laughing-q** [comment]
@jiyuwangbupt hi I want to know the loss you said is val loss or train loss?

**Comment by Laughing-q** [comment]
@jiyuwangbupt also actually I reproduced the `UserWarning` you mentioned with the code of this PR #279 , and my training and val losses look good.

**Comment by mehran66** [comment]
I see the same warning running training on coco:
nohup yolo task=detect mode=train model=yolov8x.pt epochs=300 batch=64 cache=True workers=20 name='pt_od_4' device=\'0,1,2,3,4,5,6,7\' > ./runs/pt_od_4.out &

**Comment by jiyuwangbupt** [comment]
> @jiyuwangbupt hi I want to know the loss you said is val loss or train loss?
> 
> EDIT: if it's val loss, then this should be solved in #279 which we'll merge it to main later today.

training loss
![image](https://user-images.githubusercontent.com/37127150/212224176-a73938d7-3abd-4839-a00b-6f96aa8786c1.png)

**Comment by Laughing-q** [comment]
@mehran66 @jiyuwangbupt hey guys, can you try to replace this line to `self.optimizer.step()`? and check if the losses are good. Thanks!
[https://github.com/ultralytics/ultralytics/blob/2bc36d97ce7f0bdc0018a783ba56d3de7f0c0518/ultralytics/yolo/engine/trainer.py#L410](https://github.com/ultralytics/ultralytics/blob/2bc36d97ce7f0bdc0018a783ba56d3de7f0c0518/ultralytics/yolo/engine/trainer.py#L410)

**Comment by RuoLv** [comment]
> @mehran66 @jiyuwangbupt hey guys, can you try to replace the following line to `self.optimizer.step()`? and restart training to check if the losses are good. Thanks!
> 
> https://github.com/ultralytics/ultralytics/blob/2bc36d97ce7f0bdc0018a783ba56d3de7f0c0518/ultralytics/yolo/engine/trainer.py#L410

It still not works for me.  I have the same issue as this. And use the datasets, which is directly download from roboflow, his structure have some differences from previous yolo version.

**Comment by RuoLv** [comment]
> > @mehran66 @jiyuwangbupt hey guys, can you try to replace the following line to `self.optimizer.step()`? and restart training to check if the losses are good. Thanks!
> > https://github.com/ultralytics/ultralytics/blob/2bc36d97ce7f0bdc0018a783ba56d3de7f0c0518/ultralytics/yolo/engine/trainer.py#L410
> 
> It still not works for me. I have the same issue as this. And use the datasets, which is directly download from roboflow, his structure have some differences from previous yolo version.

My env:
(yolo) C:\Users\Yewen\Desktop\workspace\yolo\rok>yolo checks  
Ultralytics YOLOv8.0.11  Python-3.10.9 torch-1.13.1+cu116 CUDA:0 (NVIDIA GeForce RTX 2070 Super, 8192MiB)
Setup complete  (16 CPUs, 31.8 GB RAM, 141.6/934.7 GB disk)

**Comment by GraBerry** [comment]
I have the same problem,but it look like harmless, I don't know what impact it will have,please tell me when you get result

**Comment by glenn-jocher** [comment]
@GraBerry @zhu000 this appears sometimes in certain versions of torch, but it's just a warning and you can ignore it. It's not showing up in later versions of torch usually, i.e. you can see our Colab notebook trianing with torch==1.13.1 does not show the warning and torch==2.0.0 doesn't either:
https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb


<img width="1441" alt="Screenshot 2023-03-26 at 15 25 21" src="https://user-images.githubusercontent.com/26833433/227778708-b868456c-c818-4edb-894a-d6eede03c444.png">

**Comment by AdamMayor2018** [comment]
tried 1.13 , shows the exact same warning.

**Comment by glenn-jocher** [comment]
@AdamMayor2018 @zhu000 This warning can be safely ignored as it does not affect the training process. It appears in some older versions of Pytorch, but it is not showing up in later versions like torch==1.13.1 or torch==2.0.0. So, if you have the latest version of Pytorch, you should not see this warning.

**Comment by YIN319** [comment]
I have the same problem, try this（amp=False）

**Comment by YIN319** [comment]
我有相同的问题，试试这样（ultralytics\yolo\cfg\default.yaml    amp: False  )

**Comment by YIN319** [comment]
I have the same problem, try this（ultralytics\yolo\cfg\default.yaml amp: False)

**Comment by glenn-jocher** [comment]
@YIN319 thank you for reporting your issue. The issue you are experiencing is related to the `lr_scheduler` and `optimizer` order, which can sometimes show up as a warning in PyTorch versions prior to 1.1.0. However, this warning does not affect the training process and can be safely ignored.

One possible workaround to suppress the warning is to set `amp: False` in your `ultralytics\yolo\cfg\default.yaml` file.

Please let us know if you have any further questions or concerns.

**Comment by glenn-jocher** [comment]
@apiszcz thank you for raising this issue. It appears that you are experiencing the `lr_scheduler` warning in PyTorch. This warning can safely be ignored as it does not affect the training process.

However, if the warning is bothersome, one possible workaround is setting `amp: False` in your `ultralytics\yolo\cfg\default.yaml` file to suppress the warning.

Please let us know if you have any further questions or concerns related to this issue, and we will be happy to assist you further.
